src/extensions/crypt/aes/aes.c

```diff
@@ -200,7 +200,6 @@ void AES_set_key(AES_CTX *ctx, const uint8_t *key,
             return;
     }
 
-    ctx->key_mode = AES_MODE_ENCRYPT; //default mode
```

Comment:
via plain 'ol process of elimination, it seems that beyond the other thing, this `ctx->key_mode` setting (here and for AES_MODE_DECRYPT) appears needed to make `read https://example.com` work.

Go figure.